jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -181,15 +181,19 @@ def _parse_esa_s3d(app, hdulist, data_label, ext='DATA', viewer_name='flux-viewe
         flux = hdulist[ext].data << unit
 
     hdr = hdulist[1].header
+
     wcs_dict = {
-        'CTYPE1': 'RA---TAN', 'CUNIT1': 'deg', 'CDELT1': hdr['CDELT2'], 'CRPIX1': hdr['CRPIX2'],
-        'CRVAL1': hdr['CRVAL2'], 'NAXIS1': hdr['NAXIS2'],
+        'CTYPE1': 'WAVE    ', 'CUNIT1': 'um', 'CDELT1': hdr['CDELT3'] * 1E6,
+        'CRPIX1': hdr['CRPIX3'] - 1,
+        'CRVAL1': hdr['CRVAL3'] * 1E6, 'NAXIS1': hdr['NAXIS3'] - 2,
```

Comment:
The shifts of 1 in CRPIX1 and NAXIS1 may be an artifact of my trimming the cube in the notebook version of the parser.  This should be reconsidered, as it may throw the wavelength axis off by one pixel.

Reply:
Ok, I changed `'CRPIX1': hdr['CRPIX3'] - 1`  to  `'CRPIX1': hdr['CRPIX3']` and `'NAXIS1': hdr['NAXIS3'] - 2,` to `'NAXIS1': hdr['NAXIS3']`, please let me know if it looks correct.

Reply:
Good